File: leetcode-submissions/881_Boats_to_Save_People.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) {
        //now sorting this vector and we need to use the two pointer approach for solving this question 
        sort(people.begin(),people.end());
        int left=0;
        int right=people.size()-1;
        int boats=0;

        while(left<=right){
            if(people[left]+people[right]<=limit){
boats++;
        left++;
        right--;
            }
        else{
            right--;
            boats++;
        }
        }
return boats;
    }
};

File: leetcode-submissions/881_Boats_to_Save_People.ts
function numRescueBoats(nums: number[], limit: number): number {
    let i :number =0 , j:number = nums.length-1, ans:number =0 ;
    nums.sort((a, b)=>a-b);
    
    while(i<=j){
        // we have three cases 
        if(i==j && nums[i]<=limit){
            ans++;
            break;
        }else if(nums[i]+nums[j]<=limit){
            ans++;
            i++;
            j--;
        }else if(nums[i]+nums[j]>limit){
            ans++;
            j--;
        }else{
            break;
        }
    }
    return ans ;
};